pkg/allocator/allocator.go

```diff
@@ -700,6 +701,11 @@ func (a *Allocator) Allocate(ctx context.Context, key AllocatorKey) (idpool.ID,
 	boff := a.backoffTemplate
 	boff.Name = key.String()
 
+	// Wait an initial (jittered) time to try and prevent multiple
+	// agents colliding on same ID when lots of pods are started
+	// with the same set of labels
+	boff.Wait(ctx)
```

Comment:
This is not needed (and could be avoided) when running in kvstore mode, because there you cannot have duplicates thanks to locking. I'm also wondering whether it could make sense to move it to the first loop iteration, to skip the wait if the identity is already known locally.

Reply:
Hmm, good point. It will be a bit awkward; it would make sense to check the local cache both before and after waiting, as another agent may have won the race.

I also think that force-waiting for 20 ms is silly; it should be waiting some time between [0,20] ms not [20 ± k] ms